rust/lance/src/index/vector/ivf/io.rs

```diff
@@ -274,12 +275,30 @@ pub(super) async fn write_hnsw_index_partitions(
         }
     }
 
+    // TODO: make it configurable, limit by the number of CPU cores & memory
+    let parallel_limit = 3;
     let mut aux_ivf = IvfData::empty();
+    let mut task_queue = VecDeque::with_capacity(parallel_limit);
+    let mut hnsw_metadata = Vec::with_capacity(ivf.num_partitions());
+    let shared_params = Arc::new(hnsw_params.clone());
     for part_id in 0..ivf.num_partitions() as u32 {
-        let start = Instant::now();
+        if task_queue.len() >= parallel_limit {
```

Comment:
There's no `else` here. What happens when there are 2 things left in the `task_queue`?

Reply:
I guess we drain the whole thing at the end.

Reply:
https://github.com/lancedb/lance/pull/2080/files#diff-67f09b41fc672bbfc51a418b542b64b1139d0f80be3e94a82d4a1eb5dec8d959R343
here it will process the left tasks